plugins/builds/index.js

```diff
@@ -294,7 +294,7 @@ const buildsPlugin = {
 
                         //   if nextBuild is stage teardown, just return nextBuild
                         if (current.stage) {
-                            const buildDeletePromises = handleStageFailure({
+                            const buildDeletePromises = await handleStageFailure({
```

Comment:
Stage teardown builds are created in `handleStageFailure` method, but that promise isn't returned.
So we should await here.